core/primitives/src/congestion_info.rs

```diff
@@ -269,10 +248,113 @@ impl CongestionInfo {
     }
 }
 
-impl CongestionInfoV1 {
+/// The extended congestion info contains the congestion info and extra
+/// information extracted from the block that is needed for congestion control.
+///
+/// It has simpler interface and it should be used instead of using the
+/// [`CongestionInfo`] directly.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct ExtendedCongestionInfo {
+    congestion_info: CongestionInfo,
+    missed_chunks_count: u64,
+}
```

Comment:
I like that abstraction! Seems like a great solution to strap on the extra context that is "external" to the congestion info but still is relevant for congestion control decisions.
It's also extensible, if perhaps we want to throttle workload based on other conditions. (like block delays, long finality, upcoming protocol changes with costly migrations, ...)